src/schedule/ucc_schedule.c

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+#include "ucc_schedule.h"
+#include "utils/ucc_compiler_def.h"
+
+void ucc_event_manager_init(ucc_event_manager_t *em)
+{
+    int i;
+    for (i = 0; i < UCC_EVENT_LAST; i++) {
+        em->listeners_size[i] = 0;
+    }
+}
+
+void ucc_event_manager_subscribe(ucc_event_manager_t *em, ucc_event_t event,
+                                 ucc_coll_task_t *task)
+{
+    em->listeners[event][em->listeners_size[event]] = task;
+    em->listeners_size[event]++;
+}
+
+void ucc_coll_task_init(ucc_coll_task_t *task)
```

Comment:
ucc_status_t

Reply:
again, see above. lets discuss in WG anyways.

Reply:
done